tests/middleware/test_proxy_headers.py

```diff
@@ -19,96 +20,439 @@
     from uvicorn.protocols.websockets.wsproto_impl import WSProtocol
 
 
-async def app(
+async def default_app(
     scope: Scope,
     receive: ASGIReceiveCallable,
     send: ASGISendCallable,
 ) -> None:
     scheme = scope["scheme"]  # type: ignore
-    host, port = scope["client"]  # type: ignore
-    addr = "%s://%s:%d" % (scheme, host, port)
-    response = Response("Remote: " + addr, media_type="text/plain")
+    if (client := scope["client"]) is None:  # type: ignore
+        client_addr = "NONE"
+    else:
+        host, port = client
+        client_addr = f"{host}:{port}"
+
+    response = Response(f"{scheme}://{client_addr}", media_type="text/plain")
     await response(scope, receive, send)
 
 
+def make_httpx_client(
+    trusted_hosts: str | list[str],
+    client: tuple[str, int] = ("127.0.0.1", 123),
+) -> httpx.AsyncClient:
+    """Create async client for use in test cases
+
+    Args:
+        trusted_hosts: trusted_hosts for proxy middleware
+        client: transport client to use
+    """
+
+    app = ProxyHeadersMiddleware(default_app, trusted_hosts)
+    transport = httpx.ASGITransport(app=app, client=client)  # type: ignore
+    return httpx.AsyncClient(transport=transport, base_url="http://testserver")
+
+
+def make_x_headers(for_: str | None, proto: str | None = "https") -> dict[str, str]:
+    """Make X-Forwarded-* header dict
+
+    Set any argument as `None` to exclude header.
+
+    Args:
+        for_: forwarded for value
+        proto: forwarded proto value
+    """
+    headers: dict[str, str] = {}
+    if for_ is not None:
+        headers["X-Forwarded-For"] = for_
+
+    if proto is not None:
+        headers["X-Forwarded-Proto"] = proto
+
+    return headers
```

Comment:
Can you remove this, please?

Reply:
Create the headers on each test.

Reply:
At this point it feels like you're requesting changes because stylistically you'd write it differently.

Having had to check all the test cases against the new logic and writing new tests where needed, these helper functions definitely helped me write more robust test cases.

Reply:
I think those helpers make it harder for me to read each individual test.